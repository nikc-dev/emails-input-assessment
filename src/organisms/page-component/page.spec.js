import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Page from './index';



it('should render Page Component correctly', () => {
  const component = shallow(
    <Page />,
  );
  expect(toJson(component)).toMatchSnapshot();
});

it('should add a random email to Page Component state correctly', () => {
    const component = shallow(
      <Page />,
    );
    component.instance().addRandomEmail();
    expect(component.state('emailsList')).toBeTruthy();
});

it('should add email to Page Component state correctly from input function', () => {
    const component = shallow(
      <Page />,
    );
    component.setState({emailsList:[]})
    component.instance().updateEmailsList({emailId:'test@test.com'});
    expect(component.state('emailsList')).toEqual([{"emailId": "test@test.com"}]);
    expect(toJson(component)).toMatchSnapshot();
});

it('should remove email from Page Component state correctly from remove function', () => {
    const component = shallow(
      <Page />,
    );
    component.setState({emailsList:[{"emailId": "test@test.com"}]})
    component.instance().removeEmail('test@test.com');
    expect(component.state('emailsList')).toEqual([]);
    expect(toJson(component)).toMatchSnapshot();
});

it('should show alert from Page Component', () => {
    const component = shallow(
      <Page />,
    );
    component.setState({emailsList:[{"emailId": "test@test.com"}]})
    component.instance().showAlert();
    expect(toJson(component)).toMatchSnapshot();
});
