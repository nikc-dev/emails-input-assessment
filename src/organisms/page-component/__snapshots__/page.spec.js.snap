// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add email to Page Component state correctly from input function 1`] = `
<div
  className="page"
>
  <div
    className="page__grey-container"
  >
    <p
      className="page__heading-text"
    >
      Share 
      <span
        className="page__heading-text__bold"
      >
        Project name 
      </span>
      with others
    </p>
    <EmailsInput
      emailsToRender={
        Array [
          Object {
            "emailId": "test@test.com",
          },
        ]
      }
      removeEmail={[Function]}
      updateEmailsList={[Function]}
    />
  </div>
  <div
    className="page__btn-container"
  >
    <Button
      handleClick={[Function]}
      title="Add Email"
    />
    <Button
      handleClick={[Function]}
      title="Get emails count"
    />
  </div>
</div>
`;

exports[`should remove email from Page Component state correctly from remove function 1`] = `
<div
  className="page"
>
  <div
    className="page__grey-container"
  >
    <p
      className="page__heading-text"
    >
      Share 
      <span
        className="page__heading-text__bold"
      >
        Project name 
      </span>
      with others
    </p>
    <EmailsInput
      emailsToRender={Array []}
      removeEmail={[Function]}
      updateEmailsList={[Function]}
    />
  </div>
  <div
    className="page__btn-container"
  >
    <Button
      handleClick={[Function]}
      title="Add Email"
    />
    <Button
      handleClick={[Function]}
      title="Get emails count"
    />
  </div>
</div>
`;

exports[`should render Page Component correctly 1`] = `
<div
  className="page"
>
  <div
    className="page__grey-container"
  >
    <p
      className="page__heading-text"
    >
      Share 
      <span
        className="page__heading-text__bold"
      >
        Project name 
      </span>
      with others
    </p>
    <EmailsInput
      emailsToRender={Array []}
      removeEmail={[Function]}
      updateEmailsList={[Function]}
    />
  </div>
  <div
    className="page__btn-container"
  >
    <Button
      handleClick={[Function]}
      title="Add Email"
    />
    <Button
      handleClick={[Function]}
      title="Get emails count"
    />
  </div>
</div>
`;

exports[`should show alert from Page Component 1`] = `
<div
  className="page"
>
  <div
    className="page__grey-container"
  >
    <p
      className="page__heading-text"
    >
      Share 
      <span
        className="page__heading-text__bold"
      >
        Project name 
      </span>
      with others
    </p>
    <EmailsInput
      emailsToRender={
        Array [
          Object {
            "emailId": "test@test.com",
          },
        ]
      }
      removeEmail={[Function]}
      updateEmailsList={[Function]}
    />
  </div>
  <div
    className="page__btn-container"
  >
    <Button
      handleClick={[Function]}
      title="Add Email"
    />
    <Button
      handleClick={[Function]}
      title="Get emails count"
    />
  </div>
</div>
`;
